### PR TITLE
Improve tiny performance

### PR DESCRIPTION
**Pull request by lindexi**
I found us set the `dashes` first but we may waste it.

When we enter this code that we will reset this value.

https://github.com/AvaloniaUI/Avalonia/blob/8427e1a34bf9bbed25ddcd3a921f39ae1f30ec07/src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs#L130

https://github.com/AvaloniaUI/Avalonia/blob/8427e1a34bf9bbed25ddcd3a921f39ae1f30ec07/src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs#L135

 As a frequently used code I think this optimize is necessary.

And I fix a comment. That it should rename `renderTarget`.

https://github.com/AvaloniaUI/Avalonia/blob/8427e1a34bf9bbed25ddcd3a921f39ae1f30ec07/src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs#L106

### CODE REVIEW

**Review by grokys — APPROVED** [review]
The change looks good to me, but could you remove the comments and undo the formatting change on line 134? Comments at this low level are't really necessary I don't think.

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs [hunk 1]
```diff
@@ -127,13 +127,21 @@ public static StrokeStyle ToDirect2DStrokeStyle(this Avalonia.Media.Pen pen, Fac
...
-                dashes = pen.DashStyle?.Dashes.Select(x => (float)x).ToArray();
+                properties.DashOffset = (float) pen.DashStyle.Offset;
+                //for `pen.DashStyle?.Dashes != null` that you shouldnt add `?` in `pen.DashStyle.Dashes`
+                dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray();
             }
+
+            //If we don't enter the code above that it is null. We should set it a default value to solve the null exception.
+            if (dashes == null)
+            {
+                dashes = new float[0];
```
Comment (kronic) [inline src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: use Array.Empty<float>

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs [hunk 2]
```diff
@@ -127,13 +127,21 @@ public static StrokeStyle ToDirect2DStrokeStyle(this Avalonia.Media.Pen pen, Fac
...
                 properties.DashStyle = DashStyle.Custom;
-                properties.DashOffset = (float)pen.DashStyle.Offset;
-                dashes = pen.DashStyle?.Dashes.Select(x => (float)x).ToArray();
+                properties.DashOffset = (float) pen.DashStyle.Offset;
+                //for `pen.DashStyle?.Dashes != null` that you shouldnt add `?` in `pen.DashStyle.Dashes`
+                dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray();
             }
+
+            //If we don't enter the code above that it is null. We should set it a default value to solve the null exception.
+            if (dashes == null)
```
[the remark this answers is not in the document]
Comment (lindexi) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @grokys  I will modify.

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs (hunk 1 above)
Comment (lindexi) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @kronic  Thx.

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs [hunk 3]
```diff
@@ -127,13 +127,17 @@ public static StrokeStyle ToDirect2DStrokeStyle(this Avalonia.Media.Pen pen, Fac
...
             };
-            var dashes = new float[0];
+            float[] dashes = null;//If we new an array first that we may waste it when we use `dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray()`. As a frequently used code I think this optimize is necessary.
             if (pen.DashStyle?.Dashes != null && pen.DashStyle.Dashes.Count > 0)
             {
                 properties.DashStyle = DashStyle.Custom;
-                properties.DashOffset = (float)pen.DashStyle.Offset;
-                dashes = pen.DashStyle?.Dashes.Select(x => (float)x).ToArray();
+                properties.DashOffset = (float) pen.DashStyle.Offset;
+                dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray();
```
[the remark this answers is not in the document]
Comment (lindexi) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @kronic Good

[the remark this answers is not in the document]
Comment (lindexi) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @kronic 

            List<double> titHruxvrvaa = new List<double>()
            {
                1d,
                2d,
                3d
            };

            var traStqjq = titHruxvrvaa.Cast<float>().ToArray();//System.InvalidCastException:“Unable to cast object of type 'System.Double' to type 'System.Single'.”

            foreach (var temp in traStqjq)
            {
                Console.WriteLine(temp);
            }

[the remark this answers is not in the document]
Comment (kronic) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @lindexi 
I'm sorry, you're right.

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs [hunk 4]
```diff
@@ -127,13 +127,17 @@ public static StrokeStyle ToDirect2DStrokeStyle(this Avalonia.Media.Pen pen, Fac
                 EndCap = pen.EndLineCap.ToDirect2D(),
                 DashCap = pen.DashCap.ToDirect2D()
             };
-            var dashes = new float[0];
+            float[] dashes = null;//If we new an array first that we may waste it when we use `dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray()`. As a frequently used code I think this optimize is necessary.
```
Comment (grokys) [inline src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: Could you remove this comment too? We really don't need a comment explaining what this is doing, it's pretty obvious ;)

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs [hunk 5]
```diff
@@ -127,13 +127,17 @@ public static StrokeStyle ToDirect2DStrokeStyle(this Avalonia.Media.Pen pen, Fac
...
             if (pen.DashStyle?.Dashes != null && pen.DashStyle.Dashes.Count > 0)
             {
                 properties.DashStyle = DashStyle.Custom;
-                properties.DashOffset = (float)pen.DashStyle.Offset;
-                dashes = pen.DashStyle?.Dashes.Select(x => (float)x).ToArray();
+                properties.DashOffset = (float) pen.DashStyle.Offset;
+                dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray();
             }
+
+            //If we don't enter the code above that it is null. We should set it a default value to solve the null exception.
```
Comment (grokys) [inline src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: And this one.

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs [hunk 6]
```diff
@@ -127,13 +127,17 @@ public static StrokeStyle ToDirect2DStrokeStyle(this Avalonia.Media.Pen pen, Fac
...
                 DashCap = pen.DashCap.ToDirect2D()
             };
-            var dashes = new float[0];
+            float[] dashes = null;//If we new an array first that we may waste it when we use `dashes = pen.DashStyle.Dashes.Select(x => (float)x).ToArray()`. As a frequently used code I think this optimize is necessary.
             if (pen.DashStyle?.Dashes != null && pen.DashStyle.Dashes.Count > 0)
             {
                 properties.DashStyle = DashStyle.Custom;
-                properties.DashOffset = (float)pen.DashStyle.Offset;
-                dashes = pen.DashStyle?.Dashes.Select(x => (float)x).ToArray();
+                properties.DashOffset = (float) pen.DashStyle.Offset;
```
Comment (grokys) [inline src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: There's still a whitespace change here.

**Review by grokys — CHANGES_REQUESTED** [review]

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs (hunk 4 above)
Comment (lindexi) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @grokys Ok.

File: src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs (hunk 5 above)
Comment (lindexi) [reply src/Windows/Avalonia.Direct2D1/PrimitiveExtensions.cs]: @grokys Ok.

**Review by grokys — APPROVED** [review]

**Comment by grokys** [comment]
Thanks @lindexi !

**Pull request merged**